### PR TITLE
Fix crash on inference of ``__dict__.items()`` of an imported module

### LINKED ISSUE #1085: AttributeInferenceError: Could not find original name for items in ``Import``
### Steps to reproduce

Cannot reproduce outside of our closed-source repository yet; help appreciated on that.

### Current behavior

Pylint fails with a backtrace.

Based on what i found so far the import statement in question is `import re`, a Python builtin. If i leave nothing but that line in that file, the error persists. If i run the same version of Pylint/astroid in a completely separate repo with only one file with only that single import line, Pylint works as expected.

### Expected behavior

Since earlier versions of Pylint and astroid work as expected on the very same code base, 

### `python -c "from astroid import __pkginfo__; print(__pkginfo__.version)"` output

2.6.2

Python versions i tested with are 3.6.10 and 3.9.5. Pylint version i use is 2.9.3.

### Backtrace

```
Exception on node <ImportFrom l.52 at 0x7fbb6d0ba370> in file '/home/polesz/Verkefni/python/gorgeous-tuatara/app/app/__init__.py'
Traceback (most recent call last):
  File "/home/polesz/.cache/pypoetry/virtualenvs/app-K022m1iN-py3.9/bin/pylint", line 8, in <module>
    sys.exit(run_pylint())
  File "/home/polesz/.cache/pypoetry/virtualenvs/app-K022m1iN-py3.9/lib/python3.9/site-packages/pylint/__init__.py", line 24, in run_pylint
    PylintRun(sys.argv[1:])
  File "/home/polesz/.cache/pypoetry/virtualenvs/app-K022m1iN-py3.9/lib/python3.9/site-packages/pylint/lint/run.py", line 384, in __init__
    linter.check(args)
  File "/home/polesz/.cache/pypoetry/virtualenvs/app-K022m1iN-py3.9/lib/python3.9/site-packages/pylint/lint/pylinter.py", line 973, in check
    self._check_files(
  File "/home/polesz/.cache/pypoetry/virtualenvs/app-K022m1iN-py3.9/lib/python3.9/site-packages/pylint/lint/pylinter.py", line 1007, in _check_files
    self._check_file(get_ast, check_astroid_module, name, filepath, modname)
  File "/home/polesz/.cache/pypoetry/virtualenvs/app-K022m1iN-py3.9/lib/python3.9/site-packages/pylint/lint/pylinter.py", line 1033, in _check_file
    check_astroid_module(ast_node)
  File "/home/polesz/.cache/pypoetry/virtualenvs/app-K022m1iN-py3.9/lib/python3.9/site-packages/pylint/lint/pylinter.py", line 1170, in check_astroid_module
    retval = self._check_astroid_module(
  File "/home/polesz/.cache/pypoetry/virtualenvs/app-K022m1iN-py3.9/lib/python3.9/site-packages/pylint/lint/pylinter.py", line 1215, in _check_astroid_module
    walker.walk(ast_node)
  File "/home/polesz/.cache/pypoetry/virtualenvs/app-K022m1iN-py3.9/lib/python3.9/site-packages/pylint/utils/ast_walker.py", line 77, in walk
    self.walk(child)
  File "/home/polesz/.cache/pypoetry/virtualenvs/app-K022m1iN-py3.9/lib/python3.9/site-packages/pylint/utils/ast_walker.py", line 74, in walk
    callback(astroid)
  File "/home/polesz/.cache/pypoetry/virtualenvs/app-K022m1iN-py3.9/lib/python3.9/site-packages/pylint/checkers/imports.py", line 551, in visit_importfrom
    imported_module = self._get_imported_module(node, basename)
  File "/home/polesz/.cache/pypoetry/virtualenvs/app-K022m1iN-py3.9/lib/python3.9/site-packages/pylint/checkers/imports.py", line 808, in _get_imported_module
    return importnode.do_import_module(modname)
  File "/home/polesz/.cache/pypoetry/virtualenvs/app-K022m1iN-py3.9/lib/python3.9/site-packages/astroid/mixins.py", line 101, in do_import_module
    return mymodule.import_module(
  File "/home/polesz/.cache/pypoetry/virtualenvs/app-K022m1iN-py3.9/lib/python3.9/site-packages/astroid/scoped_nodes.py", line 686, in import_module
    return AstroidManager().ast_from_module_name(absmodname)
  File "/home/polesz/.cache/pypoetry/virtualenvs/app-K022m1iN-py3.9/lib/python3.9/site-packages/astroid/manager.py", line 205, in ast_from_module_name
    return self.ast_from_file(found_spec.location, modname, fallback=False)
  File "/home/polesz/.cache/pypoetry/virtualenvs/app-K022m1iN-py3.9/lib/python3.9/site-packages/astroid/manager.py", line 116, in ast_from_file
    return AstroidBuilder(self).file_build(filepath, modname)
  File "/home/polesz/.cache/pypoetry/virtualenvs/app-K022m1iN-py3.9/lib/python3.9/site-packages/astroid/builder.py", line 135, in file_build
    return self._post_build(module, encoding)
  File "/home/polesz/.cache/pypoetry/virtualenvs/app-K022m1iN-py3.9/lib/python3.9/site-packages/astroid/builder.py", line 155, in _post_build
    self.delayed_assattr(delayed)
  File "/home/polesz/.cache/pypoetry/virtualenvs/app-K022m1iN-py3.9/lib/python3.9/site-packages/astroid/builder.py", line 225, in delayed_assattr
    for inferred in node.expr.infer():
  File "/home/polesz/.cache/pypoetry/virtualenvs/app-K022m1iN-py3.9/lib/python3.9/site-packages/astroid/node_classes.py", line 353, in infer
    yield from self._infer(context, **kwargs)
  File "/home/polesz/.cache/pypoetry/virtualenvs/app-K022m1iN-py3.9/lib/python3.9/site-packages/astroid/decorators.py", line 136, in raise_if_nothing_inferred
    yield next(generator)
  File "/home/polesz/.cache/pypoetry/virtualenvs/app-K022m1iN-py3.9/lib/python3.9/site-packages/astroid/decorators.py", line 100, in wrapped
    res = next(generator)
  File "/home/polesz/.cache/pypoetry/virtualenvs/app-K022m1iN-py3.9/lib/python3.9/site-packages/astroid/bases.py", line 144, in _infer_stmts
    for inferred in stmt.infer(context=context):
  File "/home/polesz/.cache/pypoetry/virtualenvs/app-K022m1iN-py3.9/lib/python3.9/site-packages/astroid/node_classes.py", line 367, in infer
    for i, result in enumerate(generator):
  File "/home/polesz/.cache/pypoetry/virtualenvs/app-K022m1iN-py3.9/lib/python3.9/site-packages/astroid/decorators.py", line 136, in raise_if_nothing_inferred
    yield next(generator)
  File "/home/polesz/.cache/pypoetry/virtualenvs/app-K022m1iN-py3.9/lib/python3.9/site-packages/astroid/decorators.py", line 100, in wrapped
    res = next(generator)
  File "/home/polesz/.cache/pypoetry/virtualenvs/app-K022m1iN-py3.9/lib/python3.9/site-packages/astroid/bases.py", line 144, in _infer_stmts
    for inferred in stmt.infer(context=context):
  File "/home/polesz/.cache/pypoetry/virtualenvs/app-K022m1iN-py3.9/lib/python3.9/site-packages/astroid/node_classes.py", line 367, in infer
    for i, result in enumerate(generator):
  File "/home/polesz/.cache/pypoetry/virtualenvs/app-K022m1iN-py3.9/lib/python3.9/site-packages/astroid/decorators.py", line 136, in raise_if_nothing_inferred
    yield next(generator)
  File "/home/polesz/.cache/pypoetry/virtualenvs/app-K022m1iN-py3.9/lib/python3.9/site-packages/astroid/decorators.py", line 100, in wrapped
    res = next(generator)
  File "/home/polesz/.cache/pypoetry/virtualenvs/app-K022m1iN-py3.9/lib/python3.9/site-packages/astroid/inference.py", line 258, in infer_import
    yield self.do_import_module(self.real_name(name))
  File "/home/polesz/.cache/pypoetry/virtualenvs/app-K022m1iN-py3.9/lib/python3.9/site-packages/astroid/mixins.py", line 120, in real_name
    raise AttributeInferenceError(
astroid.exceptions.AttributeInferenceError: Could not find original name for items in <Import l.4 at 0x7fbb665a5be0>
```

### PR DESCRIPTION
**Pull request by DanielNoord**
## Steps

- [x] For new features or bug fixes, add a ChangeLog entry describing what your PR does.
- [x] Write a good description on what the PR does.

## Description

Closes #1085.

The issue was that if a previous `context` was provided it was not correctly copied. This is problematic as the `localname` will be set in this function, altering the `context` object for other inference functions that later on use or copy the same `context` object.

## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :bug: Bug fix          |

### CODE REVIEW

File: tests/unittest_inference.py [hunk 1]
```diff
@@ -1732,8 +1734,7 @@ def __init__(self):
         """
         ast = extract_node(code, __name__)
         expr = ast.func.expr
-        with pytest.raises(InferenceError):
-            next(expr.infer())
+        self.assertIs(next(expr.infer()), util.Uninferable)
```
Comment (DanielNoord) [inline tests/unittest_inference.py:1737]: This revert the behaviour of this test to what is was before:
https://github.com/PyCQA/astroid/pull/927/files#r610819685

Comment (Pierre-Sassoulas) [reply tests/unittest_inference.py:1737]: Damn the last comment seems to predict the possible InferenceError. I wonder why we merged anyway.

**Review by Pierre-Sassoulas — APPROVED** [review]
Great job analyzing the problem blindfolded, that was a fine moment in engineering.

File: tests/unittest_inference.py (hunk 1 above)
Comment (DanielNoord) [reply tests/unittest_inference.py:1737]: I also don't really understand why the behaviour of the test changed because of that change.. `astroid` can be so hard to follow with all the decorators and `context` being passed around..
